restful.module

```diff
@@ -527,6 +527,17 @@ function restful_menu_process_callback($resource_name, $version = NULL) {
     foreach ($e->getHeaders() as $header_name => $header_value) {
       drupal_add_http_header($header_name, $header_value);
     }
+
+    if ($e->getCode() < 500) {
```

Comment:
the "problem" with watchdog_exception() is that it adds the line number where exception was thrown. However for exceptions < 500, we don't really need that info, since it's not really a server error, but just the way RESTful is taking advantage of exceptions to short circuit the flow.